Mixed Models - fixes

FBartos opened the pull request:
fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/989 (after updating `afex `package to the current CRAN version)
fixes: https://github.com/jasp-stats/jasp-issues/issues/957 (after updating `stanova` package to the current GitHub version)

vandenman commented on JASP-Engine/JASP/R/MixedModelsCommon.R:
```diff
@@ -2138,7 +2143,8 @@
         adapt_delta       = options$adapt_delta,
         control           = list(max_treedepth = options$max_treedepth),
-        seed              = if (isTRUE(options[["setSeed"]])) options[["seed"]]
+        seed              = if (isTRUE(options[["setSeed"]])) options[["seed"]],
```
shouldn't we maybe just add a function in common like:
```r
.getSeedJASP <- function(options) {
  if (is.list(options) && c("setSeed", "seed") %in% names(options)) {
    if (isTRUE(options[["setSeed"]]))
      return(options[["seed"]])
  }
```
and then call this one all over? That way the seed calling is all located in one place.

FBartos replied:
added

vandenman commented:
The code looks fine and jasp-stats/INTERNAL-jasp#989 seems fixed. 
I have no idea how this PR fixes jasp-stats/jasp-issues#957 though, although that might be because that issue is rather vague... @FBartos any idea what I should check to verify that the issue is fixed?

FBartos commented:
the issue is fixed by updating the `stanova` package. There seemed to be some mishandling of the aggregated binomial models withing it. You can verify it by fitting an aggregated binomial BGLMM.

kornbrot commented:
greart. so has stanova package been fixed and updated?
is there an update link? or should I re-install?

FBartos commented:
Hi @kornbrot , please do not do anything yet. This is being fixed only in the internal version. I will let you know when the fixed version is out and publically accessible.

kornbrot commented:
Many thanks

Sent from my iPhone

On 18 Sep 2020, at 13:34, FBartos <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @kornbrot<https://github.com/kornbrot> , please do not do anything yet. This is being fixed only in the internal version. I will let you know when the fixed version is out and publically accessible.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/jasp-stats/jasp-desktop/pull/4292#issuecomment-694839957>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABF5WKRTUZ7KZXCN3PEEXV3SGNHNXANCNFSM4RRWA35A>.

TimKDJ commented on JASP-Engine/JASP/R/common.R:
```diff
@@ -3074,3 +3074,9 @@ postProcessModuleInstall <- function(moduleLibraryPath)
   #             "Ensure that the SetSeed{} QML component is present in the QML file for this analysis."))
 }
+
+.getSeedJASP <- function(options) {
```
Unit tests found a syntax error -- one `{` is not closed

FBartos replied:
ok, thanks. Fixed

FBartos committed:
Fixes MM type II plots, agg binom and improves qml input types

- fixes plots for for type II SS for PB and LRT method for (G)LMMs (in hand with updating afex)
- fixes aggregated binomial familly for BGLMM (in hand with updating stanova)
- improves qml input types (continuous for number of trials, double for test against a value in marginal means & trends)
- some typos fixes

boutinb commented:
@TimKDJ Can we merge this PR now?
